src/docs/admin/references/NotificationServiceReference.adoc

```diff
@@ -295,3 +295,204 @@ and if they have a value depending on the event origin.
+- Do the same for the permission User.Read
+- Click on "Grant admin consent for ..."
+
+image::microsoft-app-grant-admin-consent.png[align="center"]
+
+====== Create a secret
+
+To authenticate itself, the app uses a secret. It must be created on this portal.
+
+WARNING: After the secret is created, it can be read and copied once. If lost, it is definitely lost and a new secret must be created, encoded, and replaced in the Third-Party script.
+
+- If not already there, got to "App registration" and select the App
+- Click on the "Certificates & secrets" in the left menu
+- Click on "New client secret"
+- Provide a secret description and select an expiry date. +
+
+A new secret will need to be created and provided to the script after this date.
+
+IMPORTANT: Copy the secret value as it will be hidden right after. It will be encrypted in the next step.
+
+image::microsoft-copy-secret.png[align="center"]
+
+====== Encrypt the secret
+
+The last step is to encrypt the secret as we don't want to provide it in plain.
+We will also do this for the Microsoft 365 user password.
+
+An encryption tool is provided with ProActive and located in
+----
+PROACTIVE_INSTALL_FOLDER/tools
+----
+
+To encrypt the data:
+- Open a console an navigate to the tools folder of the ProActive installation
+- Execute the encrypt script with the -d option and the data as argument.
+
+Example:
+----
+encrypt -d J8o8Q~q1T54ExSZ262mjBc-zOifI3CfFwuS0GN5srcZN
+----
+
+The output is en encrypt String enclosed by the characters ENC()
+
+Output example:
+----
+ENC(RLqBfho9KuD561d51d8zd62frDF511F5FA51650hPKJGo2p+8mWodQ9C3rF5OcdVg8ea1JrGg=)
+----
+
+You will need the full output, including the enclosing ENC().
+
+===== Configuring the Third-Party script
+
+Now that we created and/or configured the App, we need to configure the script. +
+The data copied in the earlier steps is not user specific and needs to be provided directly in the script.
+
+The script is in the same location as the properties file:
+----
+PROACTIVE_INSTALL_FOLDER/dist/war/notification-service/WEB-INF/classes/microsoftTeamsMessage.groovy
+----
+
+In this script you will find this code block, starting line 55
+
+----
+// ===== Azure Configuration =====
+// The tenant ID (organization ID)
+String tenantId=""
+// Application ID (Client ID)
+String clientId = ""
+// The secret encrypted beforehand using the encrypt script found in PROACTIVE_INSTALL/tools
+String encryptedSecret = ""; // <----- starting with ENC(....)
+----
+
+If not already done, the secret needs to be encrypted as explained in the earlier steps.
+
+- Fill in the tenantId, the clientId and the encrypted secret. +
+- Save the file.
+
+===== Configuring the notification portal
+
+The remaining configuration lies in the notification portal where we need to:
+
+ . Update User Settings
+ . Create or Update subscriptions
+
+This part is explained in the user guide, section <<../user/ProActiveUserGuide.adoc#microsoft_teams,Configure Microsoft teams messages>>
```

Comment:
"For detailed steps on updating user settings and managing subscriptions, refer to the ProActive User Guide, section <<../user/ProActiveUserGuide.adoc#microsoft_teams,Configure Microsoft Teams Messages>>."

Reply:
OK